marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/ingestion/bulkIngester.sjs

```diff
@@ -25,11 +25,38 @@ var input;         // jsonDocument*
 declareUpdate();
 
 const ingest = require("/data-hub/5/builtins/steps/ingestion/default/main.sjs");
-const consts = require('/data-hub/5/impl/consts.sjs');
+const consts = require("/data-hub/5/impl/consts.sjs");
+const HubUtils = require("/data-hub/5/impl/hub-utils.sjs");
 const state = fn.head(xdmp.fromJSON(endpointState));
 
 const work = fn.head(xdmp.fromJSON(workUnit));
 
+let collections = [];
+if(work.collections != null){
+  collections = work.collections.split(',');
+}
+
+let permissionsArray = []
+
+if(work.permissions == null){
+  work.permissions = 'data-hub-common,read,data-hub-common,update'
+}
+
+permissionsArray = new HubUtils().parsePermissions(work.permissions);
+
+const headers = {};
+headers.createdOn = consts.CURRENT_DATE_TIME
+headers.createdBy = consts.CURRENT_USER
+
+if(work.sourcename != null || work.sourcetype != null){
+  const sources = [];
+  const source = {};
+  source.name = work.sourcename
+  source.datahubSourceType = work.sourcetype
```

Comment:
Pinging @rahulvudutala  and @fsnow  and @ryanjdew  about this - this is the property name we're proposing for capturing the source type. Note that this is going on the develop branch. Nothing in DHF will care about it yet though.